### ISSUE TITLE
[Feature Request] Support even shapes other than dots to indicate progress.

### ISSUE THREAD

**Issue by agent515** [duytq94/flutter-intro-slider#129]
Journey progress is shown using 'dots' on the screens. You can change the animation, color, and radius but not the shape itself. I'm requesting a feature to support different shapes in one of the two ways:
- Accept custom widgets as `selectedIndicator`, `unselectedIndicator`
- Provide a way to select other types of indicators using enum `DotIndicatorType` (`DotIndicatorType.dot`, `DotIndicatorType.rectangle`).

In short, this feature should allow us to create a journey-progress widget as shown below.
![Screenshot 2022-07-17 at 11 28 08 AM](https://user-images.githubusercontent.com/46898603/179386020-b5b1eee3-d3b9-47e5-8d93-ab4d23f0efed.png)

**Comment by duytq94** [comment]
Hi, that's a cool idea, I'll try to develop it.

**Comment by duytq94** [comment]
Implemented at version 4.0.0
https://github.com/duytq94/flutter-intro-slider#indicatorconfig-parameter
